### PR TITLE
[WIP] Cybersun Ship rebalance (actually nerf).

### PR DESCRIPTION
**Pull request by EgorDinamit**
## About The Pull Request

**Rebalances loot on the ship, adds 2 random loot spawners.**

Amount of materials they get round-start is lowered by A LOT.
2 boxes with Tier4 parts - replaced with one box containing T1 parts for protolathe, autolathe and circuit imprinter.
Adds 2 types of this ruin that doesn't contain ghost-role spawners and can't spawn with each other.
It's done to provide a solid 30% chance for ghost-role one to spawn.

## Why It's Good For The Game

Given the number of materials you get on the ship right now, you can easily finish almost every machine at 20 minutes mark.
With this update, you won't be able to simply live on the ship and use your round-start mats for 2 hours, you'll have to, at very least, mine ores around the ship and probably on asteroids far away.

Probably, this will make them actually explore space, at least to get materials.
Basically it just means no more 30-minutes mechas, cyborg armies and so on.

## Changelog
:cl: EgorDinamit
add: 2 subtypes of forgottenship ruin. Both of those don't contain ghost-role spawners. Only one type of ruin can spawn at the same time, providing a solid 30% chance for ghost-role type to spawn.
balance: Cybersun Ship now contains a lower amount of materials, R&D crate moved to high-sec part, random "gimmick" crate added: may contain useful, or useless stuff.
/:cl:

### CODE REVIEW

File: code/game/objects/items/storage/boxes.dm [hunk 1]
```diff
@@ -1174,6 +1174,18 @@
...
+/obj/item/storage/box/stockparts/cybersun
+	name = "box of emergency stock parts"
+	desc = "Contains all stock parts you might need to setup basic R&D."
+	icon_state = "syndiebox"
+
+/obj/item/storage/box/stockparts/cybersun/PopulateContents()
+	var/static/items_inside = list(
+		/obj/item/stock_parts/matter_bin/bluespace = 6,
+		/obj/item/stock_parts/manipulator/femto = 5,
+		/obj/item/reagent_containers/glass/beaker = 4)
```
Comment (ExcessiveUseOfCobblestone) [inline code/game/objects/items/storage/boxes.dm:1186]: what's with the T4 parts?

File: code/game/objects/effects/spawners/lootdrop.dm [hunk 2]
```diff
@@ -446,6 +446,73 @@
...
+	loot = list(
+				/obj/item/slime_extract/grey,
+				/obj/item/storage/box/monkeycubes,
+				/obj/item/circuitboard/machine/processor/slime,
+				/obj/item/circuitboard/machine/monkey_recycler,
+				/obj/item/circuitboard/computer/xenobiology,
+				/obj/item/stock_parts/matter_bin/bluespace,
+				/obj/item/stock_parts/matter_bin/bluespace,
+				/obj/item/stock_parts/manipulator/pico,
+				/obj/item/stock_parts/manipulator/pico,
```
Comment (ExcessiveUseOfCobblestone) [inline code/game/objects/effects/spawners/lootdrop.dm:475]: T4/T3 parts?

File: code/game/objects/items/storage/boxes.dm (hunk 1 above)
Comment (EgorDinamit) [reply code/game/objects/items/storage/boxes.dm:1186]: It's using T4 parts right now, I just reduced the amount of them.

File: code/game/objects/effects/spawners/lootdrop.dm (hunk 2 above)
Comment (EgorDinamit) [reply code/game/objects/effects/spawners/lootdrop.dm:475]: I probably can move it 1 tier down.

File: code/game/objects/items/storage/boxes.dm (hunk 1 above)
Comment (ExcessiveUseOfCobblestone) [reply code/game/objects/items/storage/boxes.dm:1186]: WHAT IS THIS RUIN 😠

Comment (EgorDinamit) [reply code/game/objects/items/storage/boxes.dm:1186]: Alright alright, it always had Deluxe stock parts boxes, same with lavaland base.
I'm okay with lowering the tier of parts, but not to the first tier.

Comment (Qustinnus) [reply code/game/objects/items/storage/boxes.dm:1186]: make it first tier

File: code/game/objects/structures/crates_lockers/crates/secure.dm [hunk 3]
```diff
@@ -49,6 +49,71 @@
...
+/obj/structure/closet/crate/secure/gear/resources
+	desc = "An old crate, possibly containing all sort of treasures!"
+	name = "old gear crate"
+
+/obj/structure/closet/crate/secure/gear/resources/PopulateContents()
+	var/min_c = 20 //Minimum amount of minerals in the stack for common minerals
+	var/max_c = 50 //Maximum amount.
+	var/min_r = 5  //Ditto for rares
+	var/max_r = 15
+	var/pickednum = rand(1, 50)
```
Comment (Qustinnus) [inline code/game/objects/structures/crates_lockers/crates/secure.dm:61]: this should use a weighted list instead

File: code/game/objects/items/storage/boxes.dm (hunk 1 above)
Comment (EgorDinamit) [reply code/game/objects/items/storage/boxes.dm:1186]: @Qustinnus 
No.

Comment (Qustinnus) [reply code/game/objects/items/storage/boxes.dm:1186]: then ill open a pr myself 😏

Comment (EgorDinamit) [reply code/game/objects/items/storage/boxes.dm:1186]: I may see a reason to lower the tier by a bit, but I don't see any reasons to set it to the lowest one.

**Comment by ExcessiveUseOfCobblestone** [comment]
Can you please put what you want out of the role [here](https://hackmd.io/mPfrb52gSYmDe8wsgvq9zA?both) so I know where I need to tell you to tone it down before it gets to the point of no return. If you can't access edit ability you may need to ask oranges for perms via discord.

The more I read about this role and see what all they get the more it comes off as a codebaby that keeps getting fed without consideration of either how it plays ingame or without consideration of the rest of the game entirely. I'd like to have it toned down to fit with what you had perhaps initially designed the role to be (or we shift it to a specific focus if you did not have anything in mind) before I just start looking into full removal (assuming admins dont config deactivate it first).

**Comment by SumFaggotPlayTester** [comment]
Can you also throw in a lower chance for it to spawn? This is a good ruin in of itself but I'm starting to see it every round. This balance is nice as well given how much I've fiddled with it given the current meta for it gets all the machines up & running way to quickly. The idea where you fly around spess looking for mats for the ship seems like the best place this can go as you'll have hostile sapient people fighting/avoiding/sabatoging spess explorers looking for similar loot.

**Comment by Tlaltecuhtli** [comment]
remove the minerals, if they want them they can go find withe ship with gps to go to lavaland or mine asteroids  on the way, no need for free shit that not even golems get

**Comment by EgorDinamit** [comment]
@SumFaggotPlayTester I'd need to change chance for every space ruin to do so.
Though I can, instead, add 1-2 ruins that won't spawn with it to lower the chance.

**Comment by Cartographer-D** [comment]
> @SumFaggotPlayTester Their main objective, gameplay-wise, is **creating a sense of threat** for >anyone who ended up **in space**, probably the same as ashwalkers for lavaland, but while >ashwalkers are technically infinite, in space you might have only 5 humans and somewhat limited >amount of cyborgs, they might create.

So if you want a sense of threat in space, here's what you can do:
A. Add a navigation console
B. Give them a reason to take from other people (Item Analyzer, DNA analyzer, objectives)
C. Make the players the threat and not their OP turrets.
D. Limit their resources so they have to leech off of others.


> @SumFaggotPlayTesterWhile in my opinion they should be allowed to **visit** the station for any >sort of **minor** antagonistic actions(such as stealing materials, non high-risk items), they mustn't >serve a role of main antagonist in any sense.
>This means they shouldn't try to kill anyone on the station, especially with no real reason. 
You say that you want them to visit without killing anybody, but their weapons are as follows:
1. Pistol with ammo
2. A combat shotgun
3. Anything they can print in the lathe.
4. A Syndicate AI module

If you don't want them to be lethal, you should supply them with non-lethal weaponry.

Also, visiting the station as antags gives them the opportunity to escalate. If someone is captured by the station, do you think players will just say "okay, they're a goner"? No. I would say that if you want them to interact just have them be a ghost role, but then they are basically what pirates are already.

My suggestion would be to double down on what makes them interesting: They are a threat in space... ONLY. This keeps them from using a ghost role to interfere with the station (which isn't encouraged) without removing the main purpose you stated. How do you do this?
A. Prevent them from being able to teleport at all or navigate via console to the station.
B. Add in their flavor text that they shouldn't go to the station.
C. Remove the AI

> @SumFaggotPlayTesterWhile They are supposed to become a serious threat to almost anyone at >40-minutes mark minimum, but at the same time, their behavior should be limited to "maximum >defense, minimum offense".

What this will amount to is a bunch of players with a lot of deadly equipment seeking a reason to use it. Defense can mean a lot of things, and I've seen players use "defense" as a means to pre-emptively bomb a roboticist making a gygax among other things.

If you really want them to be able to impact a round, maybe their actions can somehow result in support for traitors on the station (dropping them gear, generating TC, temporarily disabling the powernet, things like that.)

**Comment by TerraGS** [comment]
I'd be in favor of a bigger time gate to reaching the point of being a reasonable threat to the station. Would also like if it spawned less often to make it more of a special thing that you can't expect to be around on a regular basis and to prevent degeneracy.

**Comment by EgorDinamit** [comment]
@TerraGS, I can try to add 2 types of this ruin, tha can't spawn with each other giving you a somewhat stable 33% chance to get this specific ruin.

@Cartographer-D, by navigation console, do you mean making it the working shuttle?

Turrets are weak for anyone who got a guns, bombs or tools to avoid them.

This PR is limiting their resources enough already, but does it randomly, so you might end up with no resources at all, or a big amount of every resource.

I can't remove their ability to create teleporter and AIs, and even if I did some shit-coding to add some sort of brain trauma, I don't want to nerf their abilities into stone age.

Flavor text supports policy, it's up to headmins to edit it any time.

Teleporter also exists to help traitors, as you said, but generating TCs sounds like a questionable idea, but I would like to give them some sort of unique item that will be useful for anyone who isn't on the ship.

**Comment by senjx** [comment]
BASED ORANGES

**Comment by Shadowflame909** [comment]
Remove the lethals. Give them CQC/E-Bow and a stealth implant.

Yeah, kind of full MGS parody. But it's pretty much better then incentivizing death and doom.

**Comment by RaveRadbury** [comment]
> being a reasonable threat to the station. 

The thing is that they shouldn't be a threat to the station. This is a scavenger ghost role, not nukies lite.

**Comment by EgorDinamit** [comment]
@RaveRadbury by "Threat to the station" I pretty much meant station's crew in general.
I have to balance it to be able to win in fair fight, but lose in full sec assault, if they are not prepared.

I rarely see station actually raid the ship, I may try and add me sort of **equivalent** of "malf AI disk" - something almost useless for ghost role itself, but unique and useful for the station.

**Comment by RaveRadbury** [comment]
@EgorDinamit Raids definitely happen, and the flavortext describes them as needing to fort and get supplies. Admins have been ruling according to the flavortext.

If you're intending for them to raid the station this is drastically different from what we expect from ghost role antags. Given that it is rolling every round, this is nukies lite then, and runs a strong risk of disrupting _any_ round given that it is a _guaranteed spawn_.

**Comment by EgorDinamit** [comment]
@RaveRadbury they aren't allowed to RAID the station, but due to headmins' policy they are allowed to steal some stuff, again, without killing indiscriminately.

I will try to fix guaranteed spawn problem in this PR.

I said _**equivalent**_ of this disk. I need something ultimately useless for the ghost role, but useful for the station. 
They already spawn with syndicate documents, but this thing is useful for cargo only.

**Comment by RaveRadbury** [comment]
> I said _**equivalent**_ of this disk. I need something ultimately useless for the ghost role, but useful for the station.

Do you mean the combat upgrade module? I thought you meant like a law 0 antag for AI.

**Comment by EgorDinamit** [comment]
Yes, ashwalkers get this combat upgrade disk which is quite useless for them.
I need something useless for cybersun crew and useful for the station.

**Comment by FloranOtten** [comment]
If you're giving them both a syndicate law module, and a combat upgrade, you've effectively given them the ability to ally a malf AI. That's quite a powerful thing.

**Comment by EgorDinamit** [comment]
Can you please read what I'm writting for once?
I'm not giving them a goddamn malf ai upgrade, read again before you understand.

**Comment by FloranOtten** [comment]
>Do you mean the combat upgrade module?

>Yes, 

Well what the fuck DO you mean?

**Comment by EgorDinamit** [comment]
@FloranOtten 

> I rarely see station actually raid the ship, I may try and add me sort of **_equivalent_** of "malf AI disk" - _something almost useless for ghost role itself, but unique and useful for the station._

**_Equivalent._**

**Comment by ExcessiveUseOfCobblestone** [comment]
DNM for 2 weeks.

**Comment by Qustinnus** [comment]
@ExcessiveUseOfCobblestone so are you doing anything as an alternative like removing it?

**Comment by ExcessiveUseOfCobblestone** [comment]
I didn't have any plans on personally making changes to the ruin, no. why do you ask? @Qustinnus
